std/utf.d

```diff
@@ -3106,11 +3106,28 @@ enum dchar replacementDchar = '\uFFFD';
  * Iterate a range of char, wchar, or dchars by code unit.
  *
  * The purpose is to bypass the special case decoding that
- * $(XREF array,front) does to character arrays.
+ * $(XREF array,front) does for character arrays. As a result, using ranges
+ * with `byCodeUnit` can be `nothrow` while $(XREF array,front) throws when
+ * it encounters invalid Unicode sequences.
+ *
+ * $(RED Warning:) This range does no encoding, so if the string you're passing
+ * to `byCodeUnit` contains things such as Unicode combining characters or
+ * graphemes that take up more than one code unit, then this range will not
+ * encode those values into code points that would produce "correct" Unicode
+ * graphemes.
+ *
  * Params:
- *      r = input range of characters, or array of characters
+ *      r = an input range of characters, or an array of characters
  * Returns:
- *      input range
+ *      An input range with a length if $(XREF traits, isAggregateType)
+ *      is `true` for `R`. Otherwise, this returns a finite random access range
+ *      with slicing.
+ * See_Also:
+ *      Refer to the std.uni docs for reference on Unicode terminology.
+ *
+ *      If the language you're dealing with has characters that rely on
+ *      graphemes and cannot be represented in either code points or code
+ *      units, see $(XREF uni, byGrapheme).
```

Comment:
I think language shouldn't be mentioned here. Combining characters can be used in text of any language.